test/controllers/user.test.js

```diff
@@ -0,0 +1,86 @@
+const app = require('../../app'),
+  request = require('supertest');
+
+const sampleUser = { name: 'nicolas', lastName: 'L', email: 'freecandy@msn.net', password: 'abc452626' };
+const sameEmailUser = {
+  name: 'F',
+  lastName: 'Perez',
+  email: 'freecandy@msn.net',
+  password: 'abc351515'
+};
+const invalidPasswordUser = {
+  name: 'M',
+  lastName: 'M',
+  email: 'mm@cgt.com',
+  password: '123'
+};
+const mentallyChallengedUser = {
+  email: 'yes'
+};
+
+describe('users api', () => {
+  it('should create an user', done =>
+    request(app)
+      .post('/users')
+      .send(sampleUser)
+      .set('Accept', 'application/json')
+      .expect(201)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+        expect(res.body.name).toEqual(sampleUser.name);
```

Comment:
With supertest you can check body like this too
```
.expect(200, {
        id: 1,
        ...fields
      });
```

Reply:
im trying to do 
```
.expect(409, { internal_code: EMAIL_EXISTS_ERROR })
```
but this fails because it gets 
```
{ message: 'Email already in use',
   internal_code: 'email_exists_error' }
``` 
is there a way to make it check just the values that i want using this method? i dont want to add the message text because it may vary

Reply:
Mmm i don't think so, documentation mentions that regex could be used here (i never used that), try something like 
```
.expect({
          "internal_code": EMAIL_EXISTS_ERROR,
          "message": /.+/
        })
```
If don't work don't worry, it's okay check that with expect